GoogleClient/GoogleClient/Platforms/Android/GoogleClientManager.cs

```diff
@@ -113,115 +136,112 @@ public void Logout()
-                            break;
-						case GoogleSignInStatusCodes.SignInFailed:
-							errorEventArgs.Error = GoogleClientErrorType.SignInFailedError;
-							errorEventArgs.Message = GoogleClientBaseException.SignInFailedErrorMessage;
-							_loginTcs.TrySetException(new GoogleClientSignInFailedErrorException());
-                            break;
-						case GoogleSignInStatusCodes.SignInCancelled:
-                            errorEventArgs.Error = GoogleClientErrorType.SignInCanceledError;
-                            errorEventArgs.Message = GoogleClientBaseException.SignInCanceledErrorMessage;
-                            _loginTcs.TrySetException(new GoogleClientSignInCanceledErrorException());
-                            break;
-                        default:
-                            errorEventArgs.Error = GoogleClientErrorType.SignInDefaultError;
-							errorEventArgs.Message = result.Status.StatusMessage;
-							_loginTcs.TrySetException(new GoogleClientBaseException(string.IsNullOrEmpty(result.Status.StatusMessage) ? GoogleClientBaseException.SignInDefaultErrorMessage : result.Status.StatusMessage));
-							break;
-					}
-
-					_onError?.Invoke(CrossGoogleClient.Current, errorEventArgs);
-                }
-
+                return;
             }
+            
+            GoogleSignInResult result = Auth.GoogleSignInApi.GetSignInResultFromIntent(data);
 
-        }
+            // Log the result of the authentication
+            Debug.WriteLine(Tag + ": Is the user authenticated? " + result.IsSuccess);
 
-        public void OnConnected(Bundle connectionHint)
-        {
-            Debug.WriteLine(Tag + ": Connection to the client succesfull");
+            if (result.IsSuccess)
+            {
+                OnSignInSuccessful(result);
+                return;
+            }
+            
+            OnSignInFailed(result);
         }
 
-        public void OnConnectionSuspended(int cause)
+        public void OnConnected(Bundle connectionHint) => Debug.WriteLine(Tag + ": Connection to the client successfull");
+
+        public void OnConnectionSuspended(int cause) => Debug.WriteLine(Tag + ": Connection to the client was suspended");
+
+        public void OnConnectionFailed(ConnectionResult result) => Debug.WriteLine(Tag + ": Connection to the client failed with error <" + result.ErrorCode + "> " + result.ErrorMessage);
+
+        private static void OnSignInSuccessful(GoogleSignInResult result)
         {
-            Debug.WriteLine(Tag + ": Connection to the client was suspended");
+            GoogleSignInAccount userAccount = result.SignInAccount;
+            GoogleUser googleUser = new GoogleUser
+            {
+                Name = userAccount.DisplayName,
+                Email = userAccount.Email,
+                Picture = new Uri((userAccount.PhotoUrl != null ? $"{userAccount.PhotoUrl}" : $"https://autisticdating.net/imgs/profile-placeholder.jpg"))
```

Comment:
Would be better to use a local placeholder or that the user specifies the placeholder.

Reply:
@rdelrosario  Can we create Separate Issue with Feature label for this?
as it was already in project not related to this pull request.